app/controllers/line_items_controller.rb

```diff
@@ -1,52 +1,59 @@
+# frozen_string_literal: true
+
 class LineItemsController < ApplicationController
-  before_action :find_item, only: %i[ destroy add_quantity reduce_quantity]
+  before_action :find_item, only: %i[destroy add_quantity reduce_quantity]
 
-  def create       
+  def create
     chosen_product = Product.find(params[:product_id])
     current_cart = @current_cart
     if current_cart.products.include?(chosen_product)
-      @line_item = current_cart.line_items.find_by(:product_id => chosen_product)
+      @line_item = current_cart.line_items.find_by(product_id: chosen_product)
       @line_item.quantity += 1
     else
       @line_item = LineItem.new
       @line_item.cart = current_cart
       @line_item.product = chosen_product
-      @line_item.quantity = 1      
+      @line_item.quantity = 1
+    end
+    begin
+      @line_item.save!
```

Comment:
What is the need to use save!.
Can't we use
if line_item.save
  #redirect
else
  #show error
end

Reply:
yes we can use save but just to practice exception handling I have used save!.
In next update I will change it to save

Reply:
update with save instead of save!